src/App.tsx

```diff
@@ -1,61 +1,102 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './components/types/Todo';
+
+export const App: React.FC = () => {
+  const [users] = useState(usersFromServer);
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+    setUserIdError(!selectedUserId);
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id), 0);
+
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      userId: selectedUserId,
+      completed: false,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+    setTitle('');
+    setSelectedUserId(0);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
```

Comment:
The `action` attribute in the form tag is not necessary in a React application where form submission is handled by JavaScript. Consider removing `action="/api/todos"` to avoid confusion, as the form submission is managed by the `handleSubmit` function.